### PR TITLE
lightning/parser: add more options to support LOAD DATA

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #41543

Problem Summary:

### What is changed and how it works?

LOAD DATA support below two cases while lightning don't

```
3,"a string containing a " quote","102.20"
```

```
The "BIG" boss
```

These are related to unescaped quotes in quoted or unquoted fields. So I add a option to support them.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -1556,38 +1556,39 @@ func (cli *testServerClient) runTestLoadData(t *testing.T, server *Server) {
 		dbt.MustExec("drop table if exists pn")
 	})
 
+	// TODO: disabled
 	// Test with upper case variables.
-	cli.runTestsOnNewDB(t, func(config *mysql.Config) {
```
Comment (lance6716) [inline server/server_test.go:1560]: @hawkingrei PTAL at this failed case

**Review by hawkingrei — APPROVED** [review]

File: server/server_test.go (hunk 1 above)
Comment (lance6716) [reply server/server_test.go:1560]: https://github.com/pingcap/tidb/issues/41596

File: br/pkg/lightning/mydump/csv_parser.go [hunk 2]
```diff
@@ -576,6 +593,21 @@ func (parser *CSVParser) readQuotedField() error {
 			if doubledDelimiter {
 				// consume the double quotation mark and continue
 				parser.recordBuffer = append(parser.recordBuffer, parser.quote...)
+			} else if parser.unescapedQuote {
```
Comment (gozssky) [inline br/pkg/lightning/mydump/csv_parser.go:596]: Should `"a""b"`  be parsed to `a""b` when `unescapedQuote` is `true`?

Comment (lance6716) [reply br/pkg/lightning/mydump/csv_parser.go:596]: No, the doc has an example

```
"The ""BIG"" boss"  -> The "BIG" boss
```

https://dev.mysql.com/doc/refman/8.0/en/load-data.html#load-data-file-location

Comment (gozssky) [reply br/pkg/lightning/mydump/csv_parser.go:596]: The rule is kind of strange.

**Review by gozssky — APPROVED** [review]

File: server/server_test.go (hunk 1 above)
Comment (hawkingrei) [reply server/server_test.go:1560]: LGTM

**Pull request merged**